aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -116,6 +134,25 @@ TORCH_META_FUNC(argmin)
   check_argmax_argmin(*this, "argmin", self, dim, keepdim);
 }
 
+TORCH_META_FUNC2(sum, dim_IntList)
+(const Tensor& self, IntArrayRef dim, bool keepdim, optional<ScalarType> opt_dtype) {
+  ScalarType dtype;
+  const auto& result = maybe_get_output();
+
+  if (result.defined()) {
+    dtype = opt_dtype.value_or(result.scalar_type());
```

Comment:
This made me raise an eyebrow but it is compatible with the old behavior of `get_dtype_from_result`

```
inline ScalarType get_dtype_from_result(Tensor& result, optional<ScalarType> dtype) {
  TORCH_CHECK(result.defined(), "Cannot create a new tensor inside a reduction op. You likely tried to call an operator with an out argument but the out argument was an undefined tensor.");
  if (dtype.has_value()) {
    return dtype.value();
  } else {
    return result.scalar_type();
  }
}

```